pod/pod.go

```diff
@@ -74,3 +74,39 @@ func (s *Service) Delete(ctx context.Context, name string) error {
 
 	return nil
 }
+
+// DelteAll deletes all pods.
+func (s *Service) DeleteAll(ctx context.Context) error {
+	pods, err := s.List(ctx)
+	if err != nil {
+		return fmt.Errorf("list pods: %w", err)
+	}
+
+	// if scheduled, delete it
+	var wg sync.WaitGroup
+	errCh := make(chan error, 1)
+	defer close(errCh)
+
+	for _, pod := range pods.Items {
+		wg.Add(1)
+		go func(pod corev1.Pod) {
+			defer wg.Done()
+			if pod.Spec.NodeName != "" {
+				err := s.Delete(ctx, pod.Name)
+				if err != nil {
+					errCh <- err
+				}
+			}
+		}(pod)
+	}
+
+	wg.Wait()
```

Comment:
I think you can delete all scheduled Pods by calling `DeleteCollection` once with `FieldSelector` in `metav1.ListOptions`.
https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/

Reply:
`spec.nodename!=""` or `spec.nodename!=null`, maybe.